test/functional/plugin/lsp/structures_spec.lua

```diff
@@ -0,0 +1,174 @@
+local helpers = require('test.functional.helpers')(after_each)
+local setpos = helpers.funcs.setpos
+local getpos = helpers.funcs.getpos
+local insert = helpers.insert
+local clear = helpers.clear
+local command = helpers.command
+
+local funcs = helpers.funcs
+local eq = helpers.eq
+local dedent = helpers.dedent
+
+describe('structures.lua', function()
+  local old_file = dedent([[
+  First line of text
+  Second line of text
+  Third line of text
+  Fourth line of text
+  ]])
+
+  local current_file = dedent([[
+    Line of text 1
+    Line of text 2
+    Line of text 3]])
+
+  before_each(function()
+    clear()
+    command('let g:structure_test = v:null')
+    insert(old_file)
+    command('new')
+    insert(current_file)
+  end)
+
+  local require_string = "require('lsp.structures')"
```

Comment:
I mentioned in the original PR that we should avoid hyper-granularity in the module hierarchy. Why should anyone need to do `require('lsp. structures'); require('lsp.callbacks'); ...` instead of just `require('lsp')`? It should be one module.

Reply:
Note `vim.lsp.structures` already works, user code never needs any "require". I'm not against merging more modules though :)